pkg/volume/gce_pd/gce_util.go

```diff
@@ -121,56 +127,34 @@ func (gceutil *GCEDiskUtil) CreateVolume(c *gcePersistentDiskProvisioner) (strin
 		return "", 0, nil, "", fmt.Errorf("the 'zone' and 'zones' StorageClass parameters must not be used at the same time")
 	}
 
-	if replicationType == replicationTypeRegionalPD && zonePresent {
-		// If a user accidentally types 'zone' instead of 'zones', we want to throw an error
-		// instead of assuming that 'zones' is empty and proceed by randomly selecting zones.
-		return "", 0, nil, "", fmt.Errorf("the '%s' replication type does not support the 'zone' parameter; use 'zones' instead", replicationTypeRegionalPD)
-	}
-
 	// TODO: implement PVC.Selector parsing
 	if c.options.PVC.Spec.Selector != nil {
 		return "", 0, nil, "", fmt.Errorf("claim.Spec.Selector is not supported for dynamic provisioning on GCE")
 	}
 
-	switch replicationType {
-	case replicationTypeRegionalPD:
-		err = createRegionalPD(
-			name,
-			c.options.PVC.Name,
-			diskType,
-			configuredZones,
-			requestGB,
-			c.options.CloudTags,
-			cloud)
-		if err != nil {
-			glog.V(2).Infof("Error creating regional GCE PD volume: %v", err)
-			return "", 0, nil, "", err
-		}
-
-		glog.V(2).Infof("Successfully created Regional GCE PD volume %s", name)
-
-	case replicationTypeNone:
+	createPD := func() error {
```

Comment:
To improve readability, move this to a new method in this file, instead of an inline anonymous function.